include/picongpu/plugins/hdf5/HDF5Writer.hpp

```diff
@@ -524,10 +524,12 @@ class HDF5Writer :
 
     void closeH5File()
     {
-        if (mThreadParams.dataCollector != nullptr)
+        if (mThreadParams.dataCollector)
         {
             log<picLog::INPUT_OUTPUT > ("HDF5 close DataCollector");
             mThreadParams.dataCollector->close();
+            mThreadParams.dataCollector->finalize();
```

Comment:
Did you RT test this?

We close files after every write. But `finalize` will [free the communicator](https://github.com/ComputationalRadiationPhysics/libSplash/blob/v1.7.0/src/ParallelDataCollector.cpp#L203-L211) that was given to the `ParallelDataCollector` in the constructor. We only generate that communicator once during a run, since it's a potentially expensive MPI operation.

We do call `finalize()` during clean shutdown in `unload()`